test/common/config/subscription_impl_test.cc

```diff
@@ -48,10 +50,20 @@ class SubscriptionImplTest : public testing::TestWithParam<SubscriptionType> {
   void initialize(std::chrono::milliseconds init_fetch_timeout = std::chrono::milliseconds(0)) {
     switch (GetParam()) {
     case SubscriptionType::Grpc:
-      test_harness_ = std::make_unique<GrpcSubscriptionTestHarness>(init_fetch_timeout);
+      test_harness_ = std::make_unique<GrpcSubscriptionTestHarness>(LegacyOrUnified::Legacy,
+                                                                    init_fetch_timeout);
       break;
     case SubscriptionType::DeltaGrpc:
-      test_harness_ = std::make_unique<GrpcSubscriptionTestHarness>(init_fetch_timeout);
+      test_harness_ = std::make_unique<GrpcSubscriptionTestHarness>(LegacyOrUnified::Legacy,
```

Comment:
I might be missing something, but shouldn't the above case just fall through to this one?

Reply:
It's a bit weird: `GrpcSubscriptionTestHarness` is used by `subscription_impl_test.cc ` and `grpc_subscription_impl_test.cc` tests (which are completely different test suites!). Both are parameterized, but in different/incompatible ways (SubscriptionType vs LegacyOrUnified), leading to this awkward implementation.